src/modules/common/UploadImage.tsx

```diff
@@ -0,0 +1,101 @@
+import { useEffect, useState } from 'react';
+
+import { Box } from '@mui/material';
+
+import { ROUTINES, useLocalContext } from '@graasp/apps-query-client';
+
+import Uppy, { UploadResult } from '@uppy/core';
+import '@uppy/core/dist/style.css';
+import '@uppy/dashboard/dist/style.css';
+import { Dashboard } from '@uppy/react';
+
+import { FILE_UPLOAD_MAX_FILES } from '@/config/constants';
+import { useNameFrameTranslation } from '@/config/i18n';
+import { NAME_THE_FRAME } from '@/langs/constants';
+
+import { hooks, mutations, notifier } from '../../config/queryClient';
+import { DASHBOARD_UPLOADER_ID } from '../../config/selectors';
+import configureUppy from '../../utils/uppy';
+
+type Props = {
+  onUploadComplete?: () => void;
+};
+const { uploadAppSettingFileRoutine } = ROUTINES;
+const UploadImage = ({ onUploadComplete }: Props): JSX.Element | null => {
+  const { t } = useNameFrameTranslation();
+  const { itemId, apiHost } = useLocalContext();
+  const { data: token } = hooks.useAuthToken(itemId);
+  const [uppy, setUppy] = useState<Uppy | null>(null);
+  const { mutate: onFileUploadComplete } = mutations.useUploadAppSettingFile();
+
+  const onComplete = (result: UploadResult): boolean | void => {
+    if (!result?.failed.length) {
+      onFileUploadComplete({
+        data: result.successful
+          ?.map(({ response }) => response?.body?.[0])
+          .filter(Boolean),
+      });
+      onUploadComplete?.();
+    }
+    return false;
+  };
+
+  const onUpload = (): void => {
+    notifier({ type: uploadAppSettingFileRoutine.SUCCESS });
+  };
+
+  const onError = (error: Error): void => {
+    onFileUploadComplete({ error });
+  };
+
+  const applyUppy = (): void => {
+    if (typeof token !== 'undefined') {
+      setUppy(
+        configureUppy({
+          apiHost,
+          itemId,
+          token,
+          onComplete,
+          onError,
+          onUpload,
+        }),
+      );
+    }
+  };
+
+  // update uppy configuration each time itemId changes
+  useEffect(() => {
+    applyUppy();
+
+    return () => {
+      uppy?.close();
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [itemId, token]);
```

Comment:
I think it can be interesting to have all this in a custom hook (in `UploadImage.hook.ts`)